core/src/main/scala/spire/math/Algebraic.scala

```diff
@@ -66,86 +167,1383 @@ extends ScalaNumber with ScalaNumericConversions
     case _ => unifiedPrimitiveEquals(that)
   }
 
-  override def hashCode: Int = if (isWhole && toBigInt == toLong) {
+  override def hashCode: Int = if (isWhole && isValidLong) {
     unifiedPrimitiveHashcode
   } else {
     val x = toBigDecimal(java.math.MathContext.DECIMAL64)
     x.underlying.unscaledValue.hashCode + 23 * x.scale.hashCode + 17
   }
-}
 
+  def toExprString: String = {
```

Comment:
I'll delete this. It's nice to have something like this, but this particular implementation is janky and I'd rather not support it (or test it in this PR).